### PR TITLE
Fix cmake build for gRPC_BUILD_CODEGEN=OFF

### PR DESCRIPTION
**Pull request by arunov**
Fix cmake build for gRPC_BUILD_CODEGEN=OFF

Issue:
With gRPC_BUILD_CODEGEN=OFF, we do not build the following libraries:
1. grpc++_error_details
2. grpc++_reflection
3. grpcpp_channelz

cmake does not prevent installation of these libraries based on the
flag, though, and we end up with cmake errors such as:
[grpc-1.17.1] CMake Error at CMakeLists.txt:3812 (install):
[grpc-1.17.1]   install TARGETS given target "grpc++_error_details"
which does not exist in
[grpc-1.17.1]   this directory.

Cause:
CMakeLists.txt has checks for gRPC_BUILD_CODEGEN around build but no
such checks exist for installation of these libraries.

grpc repo has all the build targets and dependencies in the
build.yaml file. For build systems supported by grpc (Makefile,
cmake, bazel, etc.), there are template files that use information
from the build.yaml file to generate the project files specific to a
given build system. These are generated by running
tools/buildgen/generate_projects.sh (Refer templates/README.md).

In case of cmake, the CMakeLists.txt file is generated from
templates/CMakeLists.txt.template. In the CMakeLists.txt.template
file, check for gRPC_BUILD_CODEGEN for libraries related to code gen
exist around the cmake-generator code for build (cc_library function)
but not around install (cc_install function).

Fix:
Fix is to add code in cc_install in templates/CMakeLists.txt.template
for generating gRPC_BUILD_CODEGEN check for libraries related to code
gen, similar to the checks found in cc_library.

After this, tools/buildgen/generate_projects.sh is run so that
CMakeLists.txt is freshly generated.

### CODE REVIEW

**Comment by arunov** [comment]
@nicolasnoble *ping* :)

**Comment by jtattermusch** [comment]
Please provide a description what this fixes and why.

**Comment by arunov** [comment]
Updated the commit message with more info.

File: CMakeLists.txt [hunk 1]
```diff
@@ -3920,13 +3920,15 @@ endif (gRPC_BUILD_CODEGEN)
 
 if (gRPC_BUILD_CODEGEN)
```
Comment (jtattermusch) [inline CMakeLists.txt]: Was this already fixed by other PR?

It looks like the `if (gRPC_BUILD_CODEGEN)` guards are already there :-)

Comment (jtattermusch) [reply CMakeLists.txt]: https://github.com/grpc/grpc/pull/18151/files

**Comment by jtattermusch** [comment]
This PR seems to be basically a dupe of https://github.com/grpc/grpc/pull/18151/files (which was already merged). Thanks anyway!

**Pull request closed without merging**